### PR TITLE
second type of 12j symbols

### PR DESCRIPTION
**Pull request by dshintani404**
I implemented the second type of 12j symbol.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -53,6 +54,17 @@ impl SignedSqrt {
     }
 }
 
+impl Add<SignedSqrt> for SignedSqrt {
+    type Output = Self;
+    fn add(self, other: Self) -> Self::Output {
+        SignedSqrt(
+            Rational::from_f64(
+                self.signed_sq().to_f64().sqrt() + other.signed_sq().to_f64().sqrt()
```
Comment (Rufflewind) [inline src/lib.rs]: One of the key guarantees of this package (and `SignedSqrt` in particular) is the absence of round-off errors associated with floating-point arithmetic. Is it possible to implement 12j symbols without floats?

File: src/lib.rs [hunk 2]
```diff
@@ -230,3 +250,44 @@ impl Wigner9j {
         }
     }
 }
+
+/// symmetrized version of second type of Wigner 12-j symbol
```
Comment (Rufflewind) [inline src/lib.rs:234]: Is there a reference that can be cited regarding the 12j symbol regarding the convention and formula adopted here?

The 12j symbol is a bit more esoteric than the others, so I worry the definition and phase convention might not be as standardized in literature.

File: tests/main.rs [hunk 3]
```diff
@@ -264,3 +264,24 @@ fn test_regge6j() {
         assert_eq!(vec[regge.index()], (*regge, value.clone()));
     }
 }
+
+#[test]
+fn test_wigner_12j_second() {
+    let w = Wigner12jSecond {
```
Comment (Rufflewind) [inline tests/main.rs]: Can the testing coverage be improved a little more? Consider either adding some more hard-coded, known cases, or add a regression test using the hash-based approach similar to `test_wigner_9j`?

**Review by Rufflewind — CHANGES_REQUESTED** [review]
Thank you for the pull request! Would you mind looking into the issues highlighted?

File: tests/main.rs (hunk 3 above)
Comment (dshintani404) [reply tests/main.rs]: OK! I will follow your approach using hash.

File: src/lib.rs (hunk 1 above)
Comment (dshintani404) [reply src/lib.rs]: OK, I will try.

File: src/lib.rs (hunk 2 above)
Comment (dshintani404) [reply src/lib.rs:234]: My implementation is based on the equation (19.3) on the following book (P.62).

A. P. Yutsis, I. B. Levinson, and V. V. Vanagas,Mathmatical Apparatus of the Theory of Angular Momentum(Israel Program for Scientific Translations, Jerusalem,1962).

Comment (Rufflewind) [reply src/lib.rs:234]: Thanks. Would you mind including this reference in the documentation?

File: tests/main.rs [hunk 4]
```diff
@@ -37,6 +37,12 @@ const W9J_HASHES: &[(i32, &str)] = &[
     (10, "479c0a020eaceff5539e2dda2200c1ab"), // 5898846
 ];
 
+const W12J_SECOND_HASHES: &[(i32, &str)] = &[
+    (0, "1133f539027171d60d87a58f56abf094"),
+    (2, "f6c1c20b5613ba1c128783dd4d3e8ff5"),
+    (4, "f6c1c20b5613ba1c128783dd4d3e8ff5"),
```
Comment (Rufflewind) [inline tests/main.rs]: Is this the correct hash?

Note that the tests are failing: https://travis-ci.org/github/Rufflewind/wigner-symbols-rs/builds/773396798

**Review by Rufflewind — CHANGES_REQUESTED** [review]
Thanks! Other than the two issues I mentioned, the rest looks good.

**Review by Rufflewind — APPROVED** [review]

**Comment by Rufflewind** [comment]
Thank you for your contribution!

**Pull request merged**